crates/tests/node-tests/check_state.rs

```diff
@@ -0,0 +1,132 @@
+use sqlx::postgres::PgPoolOptions;
+
+use anyhow::Result;
+use node::{
+    BlindedMessage, CheckStateRequest, GetKeysRequest, GetKeysetsRequest, MintQuoteRequest,
+    MintRequest, Proof, SwapRequest,
+};
+
+use node_tests::init_node_client;
+use nuts::Amount;
+use nuts::dhke::{blind_message, hash_to_curve, unblind_message};
+use nuts::nut00::secret::Secret;
+use nuts::nut01::PublicKey;
+use nuts::nut07::ProofState;
+use starknet_types::Unit;
+
+#[tokio::test]
+async fn works() -> Result<()> {
+    let mut client = init_node_client().await?;
+    let amount = Amount::from_i64_repr(32);
+
+    // MINT QUOTE
+    let mint_quote_request = MintQuoteRequest {
+        method: "starknet".to_string(),
+        amount: amount.into(),
+        unit: Unit::MilliStrk.to_string(),
+        description: None,
+    };
+    let mint_quote_response = client
+        .mint_quote(mint_quote_request.clone())
+        .await?
+        .into_inner();
+
+    // MINT
+    let keysets = client
+        .keysets(GetKeysetsRequest {})
+        .await?
+        .into_inner()
+        .keysets;
+    let active_keyset = keysets
+        .iter()
+        .find(|ks| ks.active && ks.unit == Unit::MilliStrk.as_str())
+        .unwrap();
+
+    let secret = Secret::generate();
+    let (blinded_secret, r) = blind_message(secret.as_bytes(), None)?;
+    let mint_request = MintRequest {
+        method: "starknet".to_string(),
+        quote: mint_quote_response.quote,
+        outputs: vec![BlindedMessage {
+            amount: amount.into(),
+            keyset_id: active_keyset.id.clone(),
+            blinded_secret: blinded_secret.to_bytes().to_vec(),
+        }],
+    };
+
+    let original_mint_response = client.mint(mint_request.clone()).await?.into_inner();
+
+    // check token state, now is unspent because node doesn't know about it yet
+    let ys = vec![hash_to_curve(secret.as_bytes())?.to_bytes().to_vec()];
```

Comment:
Can you do a call with multiples tokens at the same time please? So that we know for sure the ordering mechanism is working